enumerables.rb

```diff
@@ -0,0 +1,154 @@
+module Enumerable
+  def my_each
+    each do |value|
+      yield(value)
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    each do |value|
+      yield(value, i)
+      i += 1
+    end
+  end
+
+  def my_select
+    new_array = []
+    each do |value|
+      new_array.push(value) if yield(value)
+    end
+    new_array
+  end
+
+  def my_all?
+    flag = true
+    each do |value|
+      flag = yield(value)
+      break unless flag
+    end
+    flag
+  end
+
+  def my_any?
+    flag = false
+    each do |value|
+      flag = yield(value)
+      break if flag
+    end
+    flag
+  end
+
+  def my_none?
+    flag = false
+    each do |value|
+      flag = yield(value)
+      break if flag
+    end
+    !flag
+  end
+
+  def my_count
+    i = 0
+    if block_given?
+      each do |value|
+        i += 1 if yield(value)
+      end
+    else
+      i = length
+    end
+    i
+  end
```

Comment:
* [ ] `my_count` is identical and returns the same thing as ruby's [`count`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-count). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_count.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

* [ ] my_count counts the number of items in enum that are equal to item if an argument is given
```ruby
array = [7,8,9,5,1,0,2,4,0,1]
array.my_count(1) # return this error
   # ArgumentError:
        wrong number of arguments (given 1, expected 0)
      # ./enumerables.rb:51:in `my_count'
```